src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.cpp

```diff
@@ -0,0 +1,110 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2020.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Jihyung Kim, Timo Sachsenberg  $
+// $Authors: Jihyung Kim, Timo Sachsenberg $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/VISUAL/ANNOTATION/Annotation1DVerticalLineItem.h>
+#include <OpenMS/VISUAL/Spectrum1DCanvas.h>
+
+#include <QtCore/QPoint>
+#include <QtGui/QPainter>
+
+using namespace std;
+
+namespace OpenMS
+{
+
+  Annotation1DVerticalLineItem::Annotation1DVerticalLineItem(const double& x, const QColor& color, const QString & text) :
+      Annotation1DItem(text),
+      x_(x),
+      color_(color)
+  {
+  }
+
+  Annotation1DVerticalLineItem::Annotation1DVerticalLineItem(const Annotation1DVerticalLineItem & rhs) :
+      Annotation1DItem(rhs)
+  {
+    x_ = rhs.x_;
+    color_ = rhs.color_;
+  }
+
+  Annotation1DVerticalLineItem::~Annotation1DVerticalLineItem()
+  {
+  }
+
+  void Annotation1DVerticalLineItem::draw(Spectrum1DCanvas * const canvas, QPainter & painter, bool flipped)
```

Comment:
coding convention: put `&`, `*` etc next to the type, e.g. `QPainter&`.
Applies to a lot of places here. Should be automatically fixable with clang-format if you happen to have that available.